falcon/util/streams.py

```diff
@@ -0,0 +1,276 @@
+import functools
+import io
+
+DEFAULT_CHUNK_SIZE = io.DEFAULT_BUFFER_SIZE * 4
+
+
+class BufferedStream:
```

Comment:
My undercover plan was to phase out `BoundedStream` (in favour of this class) at some point, since this class is supposed to be its superset (backed by Cython on some platforms).

I can rename it to `BufferedReader`, that makes sense.

Reply:
Done renaming.